src/main/rx/expresslrs_common.c

```diff
@@ -213,7 +213,7 @@ void fhssGenSequence(const uint32_t uidSeed, const elrsFreqDomain_e dom)
         // if it's not the sync channel
         if (i % fhssConfig->freqCount != 0) {
             uint8_t offset = (i / fhssConfig->freqCount) * fhssConfig->freqCount; // offset to start of current block
-            uint8_t rand = rngN(fhssConfig->freqCount - 1) + 1; // random number between 1 and numFreqs
+            uint8_t rand = rngN(MAX(fhssConfig->freqCount - 1, 0)) + 1; // random number between 1 and numFreqs
```

Comment:
What is this change doing here?

Reply:
The connection between the ExpressLRS frequency hopping bug and the RC smoothing issue is indirect but real:

The ExpressLRS frequency hopping bug could cause irregular RX timing or dropouts
Irregular RX timing affects the RC smoothing calculations and filter initialization
When RC smoothing is disabled, the system becomes more sensitive to these timing irregularities
The RC smoothing filter function was being called regardless of whether smoothing was enabled, potentially using uninitialized filters
This resulted in invalid setpoint values when smoothing was disabled